flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -1635,14 +1614,12 @@ def fir_ArrayCoorOp : fir_Op<"array_coor", [NoSideEffect]> {
     auto arrTy = eleTy.dyn_cast<fir::SequenceType>();
     if (!arrTy)
       return emitOpError("must be a reference to an array");
-    auto shape = getShape();
-    if (!shape)
-      return emitOpError("must have a shape");
-    auto shapeTy = shape.getType();
+    auto shapeOp = shape();
+    auto shapeTy = shapeOp.getType();
```

Comment:
shape may be optional now, right?

Reply:
yes and looking at the code generated by table gen we don't need to check for shape and slice types as they are part of the specification already so I will just clean this up.